zapier/babel-preset-zapier
# Merge pull request #32 from zapier/bump-preset-env-version

# Description

This PR bumps `@babel/preset-env` so we can start using a couple really awesome new features in zapier/zapier:

## Optional Chaining

[Quoting from this blog](https://2ality.com/2019/07/optional-chaining.html)

The following kinds of optional operations exist.
```js
obj?.prop     // optional static property access
obj?.[«expr»] // optional dynamic property access
func?.(«arg0», «arg1») // optional function or method call
```
The rough idea is:

If the value before the question mark is neither undefined nor null, then perform the operation after the question mark.
Otherwise, return undefined.

## Nullish coalescing

[Quoting from this blog](https://2ality.com/2019/08/nullish-coalescing.html)

The result of actualValue ?? defaultValue is:

defaultValue if actualValue is undefined or null
actualValue otherwise
Examples:
```js
> undefined ?? 'default'
'default'
> null ?? 'default'
'default'

> false ?? 'default'
false
> '' ?? 'default'
''
> 0 ?? 'default'
0
```

## Steps to test this out!

1. In the repo on your machine, run `yarn link`
2. Copy and paste what it returns to you in `zapier/zapier`: `yarn link "@zapier/babel-preset-zapier"`
3. Here's the tricky part: This branch introduces some new dependency changes that will not immediately reflect in `zapier/zapier` once you've yarn link'd. You'll need to explicitly install `@babel/preset-env@7.8.3` to simulate what will happen when we install this package from `npm`.
4. Try out some new JS syntax, run `yarn @zapier/web dev` and see that it works!

## For more info...

If you want to know more about some of the new features we'll have, please check out [this blog post](https://2ality.com/2019/12/ecmascript-2020.html)
